modules/analytics/src/backend/setup.ts

```diff
@@ -8,9 +8,9 @@ export default async (bp: typeof sdk, db: Database, interactionsToTrack: string[
 
   const removeExt = (name: string) => name?.replace(/\.flow\.json$/i, '')
 
-  process.BOTPRESS_EVENTS.on('bp_core_decision_elected', ({ channel, botId, source }) => {
+  process.BOTPRESS_EVENTS.on('bp_core_decision_elected', ({ channel, botId, source, details }) => {
     if (source === 'qna') {
-      db.incrementMetric(botId, channel, 'msg_sent_qna_count')
+      db.incrementMetric(botId, channel, 'msg_sent_qna_count', details)
     }
   })
```

Comment:
Wouldn't this count the question twice ? On the next line there is already a metric which is incremented when the content is actually sent to the user (instead of being simply elected)

Is bp_core_send_content broken ?

Reply:
`bp_core_send_content` seems to only be used with NDU:
```
  private async processEventNDU(sessionId: string, event: IO.IncomingEvent) {
    if (!event.ndu || !event.ndu.actions) {
      return
    }

    let eventProcessedCalled = false
    const processEvent = async (event: IO.IncomingEvent) => {
      if (!eventProcessedCalled) {
        eventProcessedCalled = true
        this.onAfterEventProcessed && (await this.onAfterEventProcessed(event))
      }
    }

    for (const { action, data } of event.ndu.actions) {
      if (action === 'send' && data) {
        const content = data as NDU.SendContent
        await this._sendContent(content, event)

        BOTPRESS_CORE_EVENT('bp_core_send_content', {
          botId: event.botId,
          channel: event.channel,
          source: content.source,
          details: content.sourceDetails!
        })
```

Reply:
Then, shouldn't we use `bp_core_send_content` in both places, instead of the "elected" one ? It's the same signature, and storing the same information, only duplicated

Reply:
Fixed in d4ea723